pkgs/development/python-modules/paddlepaddle/default.nix

```diff
@@ -0,0 +1,71 @@
+{ stdenv
+, lib
+, buildPythonPackage
+, fetchPypi
+, python
+, pythonOlder
+, pythonAtLeast
+, autoPatchelfHook
+, openssl_1_1
+, zlib
+, setuptools
+# runtime dependencies
+, httpx
+, numpy
+, protobuf
+, pillow
+, decorator
+, astor
+, paddle-bfloat
+, opt-einsum
+}:
+
+let
+  pname = "paddlepaddle";
+  version = "2.5.0";
+  format = "wheel";
+  pyShortVersion = "cp${builtins.replaceStrings ["."] [""] python.pythonVersion}";
+  hashAndPlatform = import ./binary-hashes.nix { pythonVersion = pyShortVersion; system = stdenv.system; };
+  src = fetchPypi ({
+    inherit pname version format;
+    dist = pyShortVersion;
+    python = pyShortVersion;
+    abi = pyShortVersion;
+  } // hashAndPlatform);
+in
+buildPythonPackage {
+  inherit pname version format src;
+
+  disabled = pythonOlder "3.9" || pythonAtLeast "3.11";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    openssl_1_1
+    zlib
+    setuptools # TODO: Why is this needed?
```

Comment:
I'm not sure why, but without this addition, the build step was failing with a "setuptools not found" kind of error. Isn't setuptools available by default in `buildPythonPackage`?

Reply:
no it depends on the `format` you choose. if you use `setuptools` format then it's available, otherwise it's not.

Reply:
But it seems that the success of this build depends on the `format` being `wheel`, as `fetchPypi` is doing something with that. Shall I leave it like this? Format wheel with setuptools as a build input?